src/libraries/Microsoft.PowerFx.Connectors/Environment/SwaggerTabularService.cs

```diff
@@ -24,41 +24,100 @@ public sealed class SwaggerTabularService : CdpTabularService
         private readonly string _connectionId;
         private IReadOnlyList<ConnectorFunction> _tabularFunctions;
         private ConnectorFunction _metadataService;
+        private ConnectorFunction _createItems;
         private ConnectorFunction _getItems;
+        private ConnectorFunction _updateItem;
+        private ConnectorFunction _deleteItem;
         private readonly IReadOnlyDictionary<string, FormulaValue> _globalValues;
+        private readonly bool _defaultDataset = false;
 
         public SwaggerTabularService(IReadOnlyDictionary<string, FormulaValue> globalValues)
-            : base(GetDataSetName(globalValues), GetTableName(globalValues))
+            : base(GetDataSetName(globalValues, out bool useDefaultDataset), GetTableName(globalValues))
         {
             _globalValues = globalValues;
-            _connectionId = TryGetString("connectionId", globalValues, out string connectorId) ? connectorId : throw new InvalidOperationException("Cannot determine connectionId.");
+            _defaultDataset = useDefaultDataset;
+            _connectionId = TryGetString("connectionId", globalValues, out string connectorId, out _) ? connectorId : throw new InvalidOperationException("Cannot determine connectionId.");
         }
 
+        public static bool IsTabular(IReadOnlyDictionary<string, FormulaValue> globalValues, OpenApiDocument openApiDocument, out string error)
+        {
+            try
+            {
+                error = null;
+                return new SwaggerTabularService(globalValues).LoadSwaggerAndIdentifyKeyMethods(new PowerFxConfig(), openApiDocument);
+            }
+            catch (Exception ex)
+            {
+                error = ex.Message;
+                return false;
+            }
+        }
+
+        // Only used for testing
+        internal static (ConnectorFunction s, ConnectorFunction c, ConnectorFunction r, ConnectorFunction u, ConnectorFunction d) GetFunctions(IReadOnlyDictionary<string, FormulaValue> globalValues, OpenApiDocument openApiDocument)
+        {
+            return new SwaggerTabularService(globalValues).GetFunctions(new PowerFxConfig(), openApiDocument);
+        }
+
+        /* Schema [GET] */
+
+        private const string MetadataServiceRegex = @"/\$metadata\.json/datasets/({[^{}]+}(,{[^{}]+})?|default)/tables/{[^{}]+}$";
```

Comment:
>default [](http://example.com/codeflow?start=103&length=7)

We can't assume this name.

Reply:
observations tend to prove the opposite
an alternative would be to have a 2 step approach
- first try with dataset param {...} (or params)
- if no function is identified, try with default

WDYT?